File: PTT-GC-API/Models/Risk/Risk.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace PTT_GC_API.Models
{
    public class Risk
    {
        [Key]
        public int Id { get; set; }

        //public int RiskNo { get; set; }
        // Risk Headers
        //public int RiskId { get; set; }
        public int InitiativeId { get; set; }
        public string RiskLevelMitigationProgress { get; set; }
        public DateTime? RegisterDate { get; set; }
        public DateTime? ApprovePeriod { get; set; }
        public string Description { get; set; }
        public string RiskFactor { get; set; }
        public string Phase { get; set; }
        // Exiting Control
        public string ExitingControl { get; set; }
        public string ImpactExitingControl { get; set; }
        public string LikelihoodExitingControl { get; set; }
        public string RiskLevelExitingControl { get; set; }
        // Mitigation Plan
        public string MitigationPlan { get; set; }
        public string ImpactMitigationPlan { get; set; }
        public string LikelihoodMitigationPlan { get; set; }
        public string RiskLevelMitigationPlan { get; set; }
        // Progress
        public string MitigationProgress { get; set; }
        public string MitigationProgressImpact { get; set; }
        public string MitigationProgressLikelihood { get; set; }
    }
}

File: PTT-GC-SPA/src/app/core/models/risk.ts
import { RiskProgressModel } from './riskProgressModel';
import { KRIModel } from './kriModel';

export interface RiskModel {
    id: string,
    approvePeriod: Date,
    description: string,
    exitingControl: string,
    impactExitingControl: string,
    impactMitigationPlan: string,
    likelihoodExitingControl: string,
    likelihoodMitigationPlan: string,
    mitigationPlan: string,
    phase: string,
    registerDate: Date,
    riskFactor: string,
    riskLevelExitingControl: string,
    riskLevelMitigationPlan: string,
    riskProgressArray: Array<RiskProgressModel>,
    kriArray: Array<KRIModel>,
    mitigationProgress: string,
    mitigationProgressImpact: string,
    mitigationProgressLikelihood: string,
    riskLevelMitigationProgress: string,
    initiativeId: string,
    addNew: string
}
